packages/mui-joy/src/AltDateWidget/AltDateWidget.tsx

```diff
@@ -0,0 +1,274 @@
+  time = false,
+  disabled = false,
+  readonly = false,
+  autofocus = false,
+  options,
+  id,
+  name,
+  registry,
+  onBlur,
+  onFocus,
+  onChange,
+  value,
+}: WidgetProps<T, S, F>) {
+  const { translateString } = registry;
+  const [lastValue, setLastValue] = useState(value);
+  const [state, setState] = useReducer((state: DateObject, action: Partial<DateObject>) => {
+    return { ...state, ...action };
+  }, parseDateString(value, time));
+
+  useEffect(() => {
+    try {
+      const stateValue = toDateString(state, time);
+      if (readyForChange(state) && stateValue !== value) {
+        // The user changed the date to a new valid data via the comboboxes, so call onChange
+        onChange(stateValue);
+      } else if (lastValue !== value) {
+        // We got a new value in the props
+        setLastValue(value);
+        setState(parseDateString(value, time));
+      }
+    } catch (e) {
+      console.log('AltDateWidget useEffect error', e);
+    }
+  }, [time, value, onChange, state, lastValue]);
+
+  const handleChange = useCallback((property: keyof DateObject, value: string) => {
+    setState({ [property]: value });
+  }, []);
+
+  const handleSetNow = useCallback(
+    (event: MouseEvent<HTMLAnchorElement>) => {
+      event.preventDefault();
+      if (disabled || readonly) {
+        return;
+      }
+      const nextState = parseDateString(new Date().toJSON(), time);
+      onChange(toDateString(nextState, time));
+    },
+    [disabled, readonly, time]
+  );
+
+  const handleClear = useCallback(
+    (event: MouseEvent<HTMLAnchorElement>) => {
+      event.preventDefault();
+      console.log('AltDateWidget handleClear', disabled, readonly);
+      if (disabled || readonly) {
+        return;
+      }
+      onChange(null);
+    },
+    [disabled, readonly, onChange]
+  );
+
+  return (
+    <>
+      <Stacks
+        id={id}
+        name={name}
+        handleChange={handleChange}
+        state={state}
+        time={time}
+        options={options}
+        disabled={disabled}
+        readonly={readonly}
+        registry={registry}
+        onBlur={onBlur}
+        onFocus={onFocus}
+        autofocus={autofocus}
+      />
+      <Stack direction='row' justifyContent='center' alignItems='center' spacing={0.5}>
+        {(options.hideNowButton !== 'undefined' ? !options.hideNowButton : true) && (
+          <Button sx={{ marginTop: '0.375rem !important' }} onClick={handleSetNow}>
+            {translateString(TranslatableString.NowLabel)}
+          </Button>
+        )}
+        {(options.hideClearButton !== 'undefined' ? !options.hideClearButton : true) && (
+          <Button sx={{ marginTop: '0.375rem !important' }} onClick={handleClear}>
+            {translateString(TranslatableString.ClearLabel)}
+          </Button>
+        )}
+      </Stack>
+    </>
+  );
+}
+
+export default AltDateWidget;
+
+const Stacks = memo(
```

Comment:
Consider exporting this as separate file? If not, move about the main `AltDateWidget`